server/src/server.ts

```diff
@@ -50,6 +52,14 @@ io.on("connection", async (socket) => {
   socket.on("PONG_PADDLE_MOVEMENT", pongPaddleMovementSocket);
   socket.on("RPS_CHOOSE_ACTION", chooseActionSocket(io));
   socket.on("ADD_REACTION", addReactionSocket(io));
+
+  socket.on("QUIT_TOURNAMENT", (tournamentCode) => {
+    if (tournamentMap.has(tournamentCode)) {
+      tournamentMap.delete(tournamentCode); // Remove the tournament from the map
+      console.log(`Tournament ${tournamentCode} has been removed.`);
```

Comment:
Tbh I am not entirely sure if I want this feature (deleting a tournment to exist) just as I can see this going wrong some how in our demo. But instead we can disconnect the host from the tournament which will allow the tournament to stay alive.

Also put this code in a separate file like the rest of the socket events, to keep the server file as simple as possible.

Reply:
> Tbh I am not entirely sure if I want this feature (deleting a tournment to exist) just as I can see this going wrong some how in our demo. But instead we can disconnect the host from the tournament which will allow the tournament to stay alive.

How would it go wrong in our demo? It's just for show and I implemented this functionality for completeness. 
We don't  want the tournament to stay alive because how would the host start the tournament? 
Once the tournament actually starts the host wont have an option to end it early,.